Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h

```diff
@@ -0,0 +1,759 @@
+
+// @todo checks that cones are different, are on seams, seam is one connected
+//       component
+// @todo Should the order of cones provided in entry matter ? Map the first cone
+//       to [-1, -1] for example ?
+
+namespace CGAL {
+
+namespace Surface_mesh_parameterization {
+
+/// \ingroup PkgSurfaceParameterizationEnums
+///
+/// Weight type used in the parameterization computation.
+///
+/// MVC weights are guaranteed to generate positive edge weights, and the parameterization
+/// is guaranteed to be injective.
+///
+/// In case the cotangent weights are used, the orbifold-Tutte embedding globally
+/// minimizes the Dirichlet energy and approximates conformal mappings.
+enum Weight_type
+{
+  Cotangent = 0,
+  Mean_value
+};
+
+/// \ingroup  PkgSurfaceParameterizationMethods
+///
+/// The class `Orbifold_Tutte_parameterizer_3` implements <em>Orbifold Tutte Planar
+/// Embeddings</em> \cgalCite{aigerman2015orbifold}.
+///
+/// This is a borderless parameterization. A one-to-one mapping is guaranteed.
+///
+/// The main function of the class `Orbifold_Tutte_parameterizer_3` is `parameterize()`,
+/// to which the user provides a `Seam_mesh` with marked edges (the seams)
+/// and a set of vertices of the mesh (the cones). The choice of cones influences
+/// the resulting parameterization, but not the choice of the seam path between these cones.
+///
+/// The example \ref Surface_mesh_parameterization/orbifold.cpp "orbifold.cpp"
+/// shows how to select cones on the input mesh and automatically construct
+/// the seams and the cones on the `Seam_mesh`.
+///
+/// \cgalModels `Parameterizer_3`
+///
+/// \tparam SeamMesh must be a `Seam_mesh`, with underlying mesh any model of `FaceListGraph` and `HalfedgeListGraph`
+/// \tparam SparseLinearAlgebraTraits_d  Traits class to solve a sparse linear system
+///
+/// \sa `CGAL::Surface_mesh_parameterization::ARAP_parameterizer_3<TriangleMesh, BorderParameterizer_3, SparseLinearAlgebraTraits_d>`
+/// \sa `CGAL::Surface_mesh_parameterization::Barycentric_mapping_parameterizer_3<TriangleMesh, BorderParameterizer_3, SparseLinearAlgebraTraits_d>`
+/// \sa `CGAL::Surface_mesh_parameterization::Discrete_authalic_parameterizer_3<TriangleMesh, BorderParameterizer_3, SparseLinearAlgebraTraits_d>`
+/// \sa `CGAL::Surface_mesh_parameterization::Discrete_conformal_map_parameterizer_3<TriangleMesh, BorderParameterizer_3, SparseLinearAlgebraTraits_d>`
+/// \sa `CGAL::Surface_mesh_parameterization::LSCM_parameterizer_3<TriangleMesh, BorderParameterizer_3>`
+/// \sa `CGAL::Surface_mesh_parameterization::Mean_value_coordinates_parameterizer_3<TriangleMesh, BorderParameterizer_3, SparseLinearAlgebraTraits_d>`
+///
+template
+<
+  typename SeamMesh,
+  typename SparseLinearAlgebraTraits_d
+#ifdef CGAL_SMP_USE_SPARSESUITE_SOLVERS
+    = Eigen_solver_traits<Eigen::UmfPackLU<Eigen_sparse_matrix<double>::EigenType> >
+#else
+    = Eigen_solver_traits<Eigen::SparseLU<Eigen_sparse_matrix<double>::EigenType> >
+#endif
```

Comment:
Do you mean that you hard-coded the use of Eigen and did not use the solver traits concept?